### ISSUE TITLE
MD077: Multi-line list items aren't being re-indented

### ISSUE THREAD

**Issue by vEnhance** [rvben/rumdl#560]
### rumdl version

0.1.66

### Installation method

Pre-built binary

### Bug description

In a file such as

```markdown
# A list with multi-line items

1. This is a list item with multiple lines.
    The second line is over-indented.
```

the wrong indentation of the second line is not changed.

(It's possible I'm misunderstanding the intended behavior of MD005 for multi-line items, and that this should properly be a feature request rather than a bug report. Feel free to reclassify or close if so.)

### Steps to reproduce

```
$ cat example.md 
# A list with multi-line items

1. This is a list item with multiple lines.
    The second line is over-indented.

$ rumdl check --no-config example.md 

Success: No issues found in 1 file (24ms)

$ rumdl version
rumdl 0.1.66
```

### Expected behavior

rumdl should fix the second line.

### Actual behavior

rumdl does not re-indent the second line.

### Configuration

```toml

```

### Operating system

Linux

### Additional context

Possibly related to #439.

**Comment by rvben** [comment]
Thanks for the detailed report, @vEnhance.

You were right that MD005 wasn't the right rule for this — it checks list marker consistency, not continuation line indentation. However, we agreed this is something rumdl should catch, so we extended **MD077** to detect over-indented tight continuation lines.

Starting in v0.1.67, your example:

```markdown
1. This is a list item with multiple lines.
    The second line is over-indented.
```

now produces:

```
MD077: Continuation line over-indented (expected 3, found 4)
```

and `rumdl fmt` auto-fixes it to 3-space indent. MkDocs flavor thresholds are also respected.
